UnboundNetworking/Lobbies/ConectionHandler.cs

```diff
@@ -0,0 +1,133 @@
+                        "H","F",
+                    };
+                    _roomOptions.CustomRoomProperties = new ExitGames.Client.Photon.Hashtable();
+                    //_roomOptions.CleanupCacheOnLeave = false; Might be needed for reconection stuffs.
+                }
+                return _roomOptions;
+            }
+        }
+        public IEnumerator ConectIfDisconected(string region = "") {
+            if(!PhotonNetwork.IsConnectedAndReady) {
+                PhotonNetwork.LocalPlayer.NickName = SteamFriends.GetPersonaName();
+                PhotonNetwork.ConnectUsingSettings();
+                PhotonNetwork.AuthValues = new AuthenticationValues($"Steam={SteamUser.GetSteamID().m_SteamID}");
+                if(region != "") {
+                    PhotonNetwork.ConnectToRegion(region);
+                } else {
+                    PhotonNetwork.ConnectToBestCloudServer();
+                }
+            }
+            Debug.Log("Coneting");
+            yield return new WaitUntil(() => isConnectedToMaster);
+            Debug.Log("Conected!");
+            Debug.Log($"region {PhotonNetwork.CloudRegion}");
+        }
+        public override void OnConnectedToMaster() {
+            isConnectedToMaster = true;
+        }
+        public override void OnDisconnected(DisconnectCause cause) {
+            isConnectedToMaster = false;
+            if(cause == DisconnectCause.ClientTimeout) {
+                //attempt reconect.
+                //TODO: figure out if this is actually feasable.
+            }
+
+            GameManager.instance.GoToMenu();
+
+        }
+
+
+        public override void OnCreatedRoom() {
+            Debug.Log(Unbound_Lobby.GetLobbyCode());
+        }
+
+        public override void OnPlayerEnteredRoom(Photon.Realtime.Player newPlayer) {
+            Debug.Log($"Player Joined: {newPlayer.NickName}  ({newPlayer.UserId})");
+        }
+
+        public override void OnRoomListUpdate(List<RoomInfo> roomList) {
+            last_known_rooms = roomList;
+            Debug.Log($"updating rooms: {roomList.Count}");
+            roomList.ForEach(r => { Debug.Log(r.Name); });
+            if(isJoiningRoom) {
+                if(roomList.Count == 0) return;
+                if(((PropertyFlags)roomList[0].CustomProperties["F"] & PropertyFlags.StrictMode) != 0 && (string)roomList[0].CustomProperties["H"] != SyncModClients.GetCompatablityHash()) {
+                    //throw some kind of error to the player. 
+                    return;
+                }
+                    
+                PhotonNetwork.JoinRoom(roomList[0].Name);
+                isJoiningRoom = false;
+            }
+        }
+
+    }
+    [HarmonyPatch(typeof(NetworkConnectionHandler), "Awake")]
+    public static class DiableVanillaNetworkConnectionHandler {
+        public static void Prefix(NetworkConnectionHandler __instance) {
+            UnityEngine.Object.DestroyImmediate(__instance);
+        }
+    }
```

Comment:
Do to the fact that it is patching an instanced method, it will *never* be null, but even if it was, unity would just ignore the Destroy call. also, there aren't going to be errors thrown from that call, so a try catch block is pointless